syntax = "proto3";

import "ngraph/op_graph/serde/ops.proto";

service Hetr {
    rpc BuildTransformer (BuildRequest) returns (BuildReply) {}
    rpc Computation (ComputationRequest) returns (ComputationReply) {}
    rpc FeedInput (FeedInputRequest) returns (FeedInputReply) {}
    rpc GetResults (GetResultsRequest) returns (GetResultsReply) {}
}

message BuildRequest {
    string transformer_type = 1;
}

message BuildReply {
    bool status = 1;
}

message ComputationRequest {
    GraphDef subgraph = 1;
    repeated Op returns = 2;
    repeated Op placeholders = 3;
}

message ComputationReply {
    int32 comp_id = 1;
}

message FeedInputRequest {
    int32 comp_id = 1;
    repeated DTYPE scalar_values = 2;
    repeated Tensor tensor_values = 3;
}

message FeedInputReply {
    bool status = 1;
}

message GetResultsRequest {
    int32 comp_id = 1;
}

message GetResultsReply {
    repeated DTYPE results = 1;
}

